Sage/carbon
# feat(portal): add inertOptOut prop to keep the content interactable when a modal opens

Recent versions of Carbon mark all outside elements with the `inert` and `aria-hidden` attributes when a modal component is opened. This makes it so that no outside content can be clicked on, or interacted with using assistive technologies such as screenreaders - and is quite deliberate because a modal usually means that nothing else in the document should be functional. However in order to cater for special cases where a small area of content is still intended to be interacted with, this commit introduces a new prop `inertOptOut` on the `Portal` component, which opts out of this behaviour. If you wrap the important content in a `Portal` with the `inertOptOut` prop, this will keep it clickable (and appearing visually on top of the modal) even when a modal is open.

### Proposed behaviour


When some UI components are wrapped in a `<Portal inertOptOut>`, they will still be able to be interacted with even when a modal (sidebar/dialog/etc) is opened elsewhere in the DOM.

### Current behaviour


Any modal opening adds the `inert` and `aria-hidden` attributes to any external UI components, making them not respond to clicks. While this is the desired behaviour in general, in Carbon currently there is no way to opt out of this behaviour.

### Checklist




#### QA


### Additional context



### Testing instructions



See this codesandbox for a demonstration: https://codesandbox.io/s/bitter-firefly-q9r5n2?file=/src/App.tsx

Note that the `inertOptOut` prop is a typescript error in master (which is what the red underline is about), but this doesn't matter - if there was code in the component to handle it it would still work, and the bottom button would fire an alert when clicked with the sidebar open. But it doesn't (and nor is there any other way in Carbon to keep it clickable when the sidebar opens).

When testing on the branch of this PR, that bottom button should respond to clicks even with the sidebar open - but the button above it still should't.
